packages/hierarchy-builder/src/hierarchy-builder/HierarchyProvider.ts

```diff
@@ -201,6 +202,19 @@ export class HierarchyProvider {
   }
 }
 
+async function createGroupingHandlers(metadata: IMetadataProvider, nodes: HierarchyNode[]): Promise<GroupingHandlerType[]> {
+  const groupingHandlers: GroupingHandlerType[] = new Array<GroupingHandlerType>();
+  const baseClassGroupingECClasses = await getBaseClassGroupingECClasses(metadata, nodes);
+  for (const baseECClass of baseClassGroupingECClasses) {
+    groupingHandlers.push(async (allNodes: HierarchyNode[]) => {
+      return createBaseClassGroupsForSingleBaseClass(metadata, allNodes, baseECClass);
+    });
+  }
```

Comment:
I think this code should be moved into `BaseClassGrouping.ts` into a function `createBaseClassGroupingHandlers(metadata, nodes)`

Reply:
Added "createBaseClassGroupingHandlers" function